R/paginate.R

```diff
@@ -86,48 +91,99 @@ paginate_req_perform <- function(req,
   check_number_whole(max_pages, allow_infinite = TRUE, min = 1)
   check_bool(progress)
 
+  the$last_pagination_request <- req
+  the$last_pagination_responses <- NULL
+  the$last_pagination_page <- NULL
+  the$last_pagination_n_pages <- NULL
+  the$last_pagination_max_pages <- max_pages
+
   resp <- req_perform(req)
   f_n_pages <- req$policies$paginate$n_pages %||% function(resp) Inf
-  n_pages <- min(f_n_pages(resp), max_pages)
+  n_pages <- f_n_pages(resp)
+  the$last_pagination_n_pages <- f_n_pages(resp)
   # the implementation below doesn't really support an infinite amount of pages
   # but 100e3 should be plenty
-  if (is.infinite(n_pages)) {
-    n_pages <- 100e3
-  }
+  n_pages <- min(the$last_pagination_n_pages, max_pages, 100e3)
 
-  out <- vector("list", length = n_pages)
-  out[[1]] <- resp
+  the$last_pagination_responses <- vector("list", length = n_pages)
+  the$last_pagination_responses[[1]] <- resp
+  the$last_pagination_page <- 1
+
+  paginate_perform_other_pages(
+    start = 2,
+    n_pages = n_pages,
+    req = req,
+    resp = resp
+  )
+}
 
+paginate_perform_other_pages <- function(start, n_pages, req, resp) {
   cli::cli_progress_bar(
     "Paginate",
     total = n_pages,
-    format = "{cli::pb_spin} Page {cli::pb_current}/{cli::pb_total} | ETA: {cli::pb_eta}",
-    current = 1L
```

Comment:
`current` is actually not the current page as I though but whether this is the current progress bar or not.